backend/build_docs.py

```diff
@@ -16,134 +17,120 @@
+    if 'OR_docs' in dst_path:
+        source_dict[dst_path] = (
+            f"{or_docs_url}/{src_path.split('_sources/')[-1].replace('.md', '.html')}"
+        )
+    elif 'ORFS_docs' in dst_path:
+        source_dict[dst_path] = (
+            f"{orfs_docs_url}/{src_path.split('_sources/')[-1].replace('.md', '.html')}"
+        )
+    elif 'manpages' in dst_path:
+        source_dict[dst_path] = (
+            f"OpenROAD Manpages - {dst_path.split('data/markdown/manpages')[-1]}"
+        )
+    elif 'yosys' in dst_path:
+        source_dict[dst_path] = f"https://{dst_path[len('data/html/') :]}"
+    elif 'OpenSTA' in dst_path and 'pdf' in dst_path:
+        source_dict[dst_path] = opensta_docs_url
+    elif 'OpenSTA' in dst_path and 'markdown' in dst_path:
+        source_dict[dst_path] = opensta_readme_url
+    elif 'theopenroadproject' in dst_path:
+        source_dict[dst_path] = (
+            f"https://{dst_path.replace('data/html/or_website/', '').replace('/index.html', '')}"
+        )
+    else:
+        source_dict[dst_path] = dst_path
+
+
 def purge_folders(folder_paths: list[str]) -> None:
     for folder_path in folder_paths:
         if os.path.exists(folder_path):
             shutil.rmtree(folder_path)
             logging.debug(f'Purging, Folder {folder_path} deleted.')
 
 
-def track_src(src: str) -> dict[str, str]:
-    copied_files = {}
-
+def track_src(src: str) -> None:
+    logging.debug(f'Updating source dict for {src}...')
     if not os.path.exists(src):
-        logging.debug(f'File {src} does not exist. Exiting.')
+        logging.error(f'File {src} does not exist. Exiting.')
         sys.exit(1)
 
     for root, _, files in os.walk(src):
         for file in files:
             src = os.path.join(root, file)
             src_path = src.split('backend/')[-1]
 
-            if 'OR_docs' in src:
-                copied_files[src_path] = (
-                    f"{or_docs_url}/{src.split('_sources/')[-1].replace('.md', '.html')}"
-                )
-            elif 'ORFS_docs' in src:
-                copied_files[src_path] = (
-                    f"{orfs_docs_url}/{src.split('_sources/')[-1].replace('.md', '.html')}"
-                )
-            elif 'manpages' in src:
-                copied_files[src_path] = (
-                    f"OpenROAD Manpages - {src_path.split('/')[-1]}"
-                )
-            elif 'yosys' in src:
-                copied_files[src_path] = src_path.split('/data/rtdocs/')[-1]
-            elif 'OpenSTA' in src:
-                copied_files[src_path] = opensta_docs_url
-            else:
-                copied_files[src_path] = src_path
-
-    return copied_files
+            update_src(src_path, src_path)
 
 
-def copy_file_track_src(src: str, dst: str) -> dict[str, str]:
-    copied_files = {}
-
+def copy_file_track_src(src: str, dst: str) -> None:
     if not os.path.exists(src):
         logging.debug(f'File {src} does not exist. Exiting.')
```

Comment:
Make it consistent: do you want to have a `error` level log before `sys.exit(1)` calls like you have above?

Reply:
Yes, that'll look better.